pycode/memilio-epidata/memilio/epidata/geoModificationGermany.py

```diff
@@ -307,7 +320,153 @@ def get_nuts3_county_id_map(merge_eisenach=True):
+
+
+def get_intermediateregion_names_and_ids(zfill=False):
+    """"! Get list of intermediate region names and IDs sorted according to ID.
+
+    @param zfill [Default: False] Defines whether IDs are zero-filled to
+        two digits and returned as a string or returned as an integer.
+    @return List of intermediate region names and IDs sorted according to state ID.
+    """
+    intermediateregionids = get_intermediateregion_ids(zfill=zfill)
+
+    return [[dd.IntermediateRegions[int(intermediateregionids[i])],
+             intermediateregionids[i]] for i in range(len(intermediateregionids))]
+
+
+def get_intermediateregion_to_name():
+    """"! Returns a hash map from federal state ID to state name.
+
+    @return hash map from federal state ID to state name.
+    """
+    return dd.IntermediateRegions
+
+
+def get_countyid_to_intermediateregionid_map(merge_eisenach=True, zfill=False):
+    """! Creates a hash map from county IDs to intermediate region IDs
+
+    @param merge_eisenach [Default: True] Defines whether the counties 
+        'Wartburgkreis' and 'Eisenach' are listed separately or combined 
+        as one entity 'Wartburgkreis'.
+    @param zfill [Default: False]. Defines whether or not all IDs are returned
+        as zero-filled strings. By default, integer maps are returned.
+    @return County ID to intermediate region ID map.
+    """
+    county_ids = get_county_ids(merge_eisenach)
+    regions_to_county = dd.IntermediateRegionIDsToCountyIDs
+
+    regions_sorted = [0 for i in range(len(county_ids))]
+    counties_sorted = [0 for i in range(len(county_ids))]
+
+    idx = 0
+    # region will be a list of region id first and a list of counties ids second
+    for region in regions_to_county.items():
+        if merge_eisenach and CountyMerging[16063][1] in region[1]:
+            region[1].remove(CountyMerging[16063][1])
+        for j in range(len(region[1])):
+            regions_sorted[idx] = region[0]
+            if zfill:
+                counties_sorted[idx] = str(region[1][j]).zfill(5)
+            else:
+                counties_sorted[idx] = region[1][j]
+
+            idx += 1
+
+    return dict(zip(counties_sorted, regions_sorted))
+
+
+def get_intermediateregionid_to_countyids_map(
+        merge_eisenach=True, zfill=False):
+    """! Creates a hash map from intermediate region IDs to lists of county IDs
+
+    @param merge_eisenach [Default: True] Defines whether the counties 
+        'Wartburgkreis' and 'Eisenach' are listed separately or combined 
+        as one entity 'Wartburgkreis'.
+    @param zfill [Default: False]. Defines whether or not all IDs are returned
+        as zero-filled strings. By default, integer maps are returned.
+    @return Intermediate region IDs to lists of county IDs map
+    """
+    regions_to_county = dd.IntermediateRegionIDsToCountyIDs
+    if zfill or merge_eisenach:
+        for i in range(len(regions_to_county)):
```

Comment:
changed loop completely

Reply:
I liked that you checked whether zfill or merge_eisennach is true before since otherwise the dictionary can be returned as it is.

Reply:
Unfortunately there was a bug in there before. The "remove" function did not remove the item based on a copy but on a reference so that a second call to this function failed. In order to correct this, also the if had to be adapted.